File: Python/Punto_1.py
from functools import reduce
#Ventas del vendedor, ingresadas por el vendedor
ventas=list(map(lambda elemento:int(input(f"Ingrese su venta {elemento+1}: ")),range(3)))
# Sueldo base del vendedor, que el vendedor no puede modificar
sueldoBase=15000
# Se suman las ventas y se obtiene el 10% para comisión
comisiones=reduce(lambda acumulador=0,elemento=0:acumulador+elemento,ventas)*0.1
# Se calcula el sueldo total sumando el sueldo base y las comisiones
sueldoTotal=sueldoBase+comisiones
# Se imprimen las comisiones y el sueldo total
print(f"Comisión: {comisiones}")
print(f"Sueldo total: {sueldoTotal}")

File: Java/Punto1.java
package Java;
import java.util.List;
import java.util.Scanner;
import java.util.stream.Collectors;
import java.util.stream.Stream;

@FunctionalInterface
interface IngresoVentas {
    List<Integer> obtenerVentas();
}

public class Punto1 {
    public static void main(String[] args) {
        double sueldobase = 500;
        IngresoVentas ingresoVentas = () -> {
            Scanner scanner = new Scanner(System.in);
            System.out.println("Ingrese los montos de ventas separados por espacios:");
            String input = scanner.nextLine();
            scanner.close();
            return Stream.of(input.split(" "))
                    .map(Integer::parseInt)
                    .collect(Collectors.toList());
        };

        List<Integer> Ventas = ingresoVentas.obtenerVentas();

        double comisiones = Ventas.stream() // Convertimos la lista en stream
                .mapToDouble(ventaporcentaje -> ventaporcentaje * 0.1) // Aplicamos el porcentaje de comisión
                .sum(); // Sumamos las comisiones

        double sueldototal = sueldobase + comisiones;

        System.out.println("Comisiones $: " + comisiones);
        System.out.println("Sueldo Total $: " + sueldototal);
    }
}